### PR TITLE
SDIT-1799 Create standard set of casenotes Nomis api endpoints

### PR DESCRIPTION
**Pull request by steverendell**
Switch from booking to prisoner based fetching

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/casenotes/CaseNotesResource.kt [hunk 1]
```diff
@@ -184,59 +181,59 @@ class CaseNotesResource(
...
+//        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+//      ),
+//    ],
+//  )
+//  fun getBookings(
+//    @Schema(description = "Start or minimum booking id", example = "12345678") @RequestParam fromId: Long?,
+//    @Schema(description = "End or maximum booking id", example = "98765432") toId: Long?,
+//    @Schema(description = "If true return only bookings currently in prison") activeOnly: Boolean,
+//    pageable: Pageable,
+//  ): Page<BookingIdResponse> = caseNotesService.getAllBookingIds(fromId, toId, activeOnly, pageable)
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/casenotes/CaseNotesResource.kt:231]: Left some commented out stuff in as this is all WIP atm

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/casenotes/CaseNotesService.kt [hunk 2]
```diff
@@ -108,22 +109,29 @@ class CaseNotesService(
   /**
    * For reconciliation or migration
    */
-  fun getCaseNotes(bookingId: Long): BookingCaseNotesResponse {
-    offenderBookingRepository.findById(bookingId).orElseThrow { NotFoundException("bookingId $bookingId not found") }
-    return BookingCaseNotesResponse(
-      offenderCaseNoteRepository.findAllByOffenderBooking_BookingId(bookingId).map { it.toCaseNoteResponse() },
+  fun getCaseNotes(offenderNo: String): PrisonerCaseNotesResponse {
+    offenderRepository.findByNomsId(offenderNo).takeIf {
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/casenotes/CaseNotesService.kt:113]: think you can just do an `ifEmpty` here instead?

**Review by petergphillips — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/casenotes/CaseNotesService.kt (hunk 2 above)
Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/casenotes/CaseNotesService.kt:113]: ah cool, ta. Will change next time

**Pull request merged**